source/frontend/src/features/mapSideBar/acquisition/tabs/compensation/detail/CompensationRequisitionDetailView.tsx

```diff
@@ -53,31 +56,37 @@ export const CompensationRequisitionDetailView: React.FunctionComponent<
   onGenerate,
 }) => {
   const { hasClaim, hasRole } = useKeycloakWrapper();
-  const getPayeeDetails = (
-    compensationPayee: Api_CompensationPayee | null | undefined,
-  ): PayeeViewDetails | null => {
-    if (!compensationPayee) {
-      return null;
+  const [payeeDetails, setPayeeDetails] = useState<PayeeViewDetails | null>(null);
+
+  useEffect(() => {
+    if (!compensation) {
+      setPayeeDetails(null);
+      return;
     }
 
-    let payeeDetail = {
+    let payeeDetail: PayeeViewDetails = {
       contactEnabled: true,
-      goodTrust: compensationPayee?.isPaymentInTrust,
-      personId: null,
-    } as PayeeViewDetails;
-
-    if (compensationPayee.acquisitionOwnerId) {
-      const ownerDetail = DetailAcquisitionFileOwner.fromApi(compensationPayee.acquisitionOwner!);
+      goodTrust: compensation?.isPaymentInTrust || false,
+      contactString: null,
+      displayName: '',
+      isGstApplicable: false,
+    };
+
+    if (!!compensation.acquisitionOwner) {
+      const ownerDetail = DetailAcquisitionFileOwner.fromApi(compensation.acquisitionOwner);
       payeeDetail.displayName = ownerDetail.ownerName ?? '';
       payeeDetail.contactEnabled = false;
-    } else if (compensationPayee.interestHolderId) {
-      payeeDetail.displayName = compensationPayee.interestHolder?.person
-        ? formatApiPersonNames(compensationPayee.interestHolder?.person)
-        : compensationPayee.interestHolder?.organization?.name ?? '';
-      payeeDetail.personId = compensationPayee.interestHolder?.person?.id!;
-    } else if (compensationPayee.motiSolicitorId) {
-      payeeDetail.displayName = formatApiPersonNames(compensationPayee.motiSolicitor);
-      payeeDetail.personId = compensationPayee.motiSolicitor?.id!;
+    } else if (compensation.interestHolderId) {
+      if (compensationContactPerson) {
+        payeeDetail.displayName = formatApiPersonNames(compensationContactPerson);
+        payeeDetail.contactString = 's' + compensationContactPerson?.id;
+      } else if (compensationContactOrganization) {
+        payeeDetail.displayName = compensationContactOrganization?.name ?? '';
+        payeeDetail.contactString = 'p' + compensationContactOrganization.id;
+      }
```

Comment:
If I understood correctly, here contactString can take the following values: "s1" or "p1" (assuming Id = 1)
But if I recall correctly, the convention for contact-based URLs are "P1" or "O1"  (for person or organization).

Could you please comment where the "s" and "p" are coming from?